### ISSUE TITLE
Clicking Buttons Does Nothing in Flatpak Wayland.

### ISSUE THREAD

**Issue by YouCanTouCan** [writingtool-org/writingtool#12]
I was having this issue with the LanguageTool extension, and am still having it with this extension as well (version 1.0). When I click on any of the buttons from the toolbar, simply nothing happens. The extension does work - it is showing me grammar and spelling errors all throughout my writing - but not being able to access the settings is frustrating. If I right-click on an underlined error, there is a WritingTool submenu in the context menu with the same options as the toolbar, but those also do nothing when clicked.

LibreOffice is installed via Flatpak on my Aurora system (an edited version of Kionite, which itself is an immutable version of Fedora). The following is my LibreOffice Version information:

```
Version: 24.8.2.1 (X86_64) / LibreOffice Community
Build ID: 0f794b6e29741098670a3b95d60478a65d05ef13
CPU threads: 32; OS: Linux 6.10; UI render: default; VCL: gtk3
Locale: en-US (en_US.UTF-8); UI: en-US
Flatpak
Calc: threaded
```

My Java version is 21.0.4, as installed by default in the Flatpak version of LibreOffice

If, through Flatseal, I turn off Wayland Windowing System suddenly the buttons do work... but LibreOffice lags like crazy, making it largely unusable. If I then disable the WritingTool extension, the lag stops.

Please let me know if there is any more info I can give. This project looks fantastic so far, and I'm still happily using it (just without access to the settings menu!) Thank you for all your hard work!

**Comment by securegh** [comment]
I have similar issue on Fedora 40 X11 windowing, LibreOffice rpm version:
Version: 24.2.7.2 (X86_64)
Build ID: 420(Build:2)
CPU threads: 8; OS: Linux 6.11; UI render: default; VCL: gtk3
Locale: en-US (en_US.UTF-8); UI: en-US

The extension settings buttons/ right click submenu are not responsive, but I am seeing text corrections.

Also, per extension description:
>AI support for automatic background checks, manual editing of paragraphs, creation of texts and illustrations (An AI server is required).
@YouCanTouCan 
Is there any documentation how to set up the extension to work with localAI?

**Comment by YouCanTouCan** [comment]
Interesting, then this may be a general Fedora issue rather than a Wayland or Flatpak issue.

I am not aware of how to set it up to work with localAI, sorry. That would be awesome to do, however, so I would also be curious.

**Comment by FredKruse** [comment]
@YouCanTouCan, @securegh 

- To install localAI look here: https://localai.io/basics/getting_started/. How to integrate it into WT can be found here: https://writingtool.org/index.php/en/help/options/ai-support-settings
- To evaluate the issue, I need to know if WritingTool is enabled as service. Look at the menu: Tools/Options /Languages and Locales/Writing Aids. WritingTool has to be listed and enabled under "Available Language Models".
If not: check under Tools/Options/Advanced if "Use a Java runtime environment" is enabled (Version >= 8 for WritingTool 1.0, Version >= 17 for WritingTool 1.1-Snapshot).

Please give me a feedback.

**Comment by FredKruse** [comment]
It seems there is a problem with the java UIManager as the log file shows. I tried to catch the error and switch back to the default system look and feel. Currently, I don't have a test environment for this system, so please test it.

**Comment by securegh** [comment]
> It seems there is a problem with the java UIManager as the log file shows. I tried to catch the error and switch back to the default system look and feel. Currently, I don't have a test environment for this system, so please test it with tomorrow's snapshot. You will find it here: https://writingtool.org/writingtool/snapshots/

That would be the 2024/11/16 snapshot? Looking forward to testing it out!

**Comment by securegh** [comment]
@BobSoft3000 as soon as @FredKruse confirms the snapshot with the fix, I would test it out on my fedora setup.

**Comment by FredKruse** [comment]
I found two problems:
1. Fedora installs openjdk-headless by default. That is not enough to run java applications with a GUI. So, please install full openjdk ( >= 17) to run WritingTool. I couldn't test all the different systems. So please ensure that full JDK is installed on your machines.
2. Some of the log files from @BobSoft3000 seem to show another problem with GTK. I have integrated a query that ensures that a GTK look-and-feel is installed on the machine (use the system look and feel otherwise). It is available from WritingTool_2024-11-17-snapshot.oxt.

**Comment by FredKruse** [comment]
@BobSoft3000 @securegh @YouCanTouCan 
Please let me know if the problem is resolved.

**Comment by YouCanTouCan** [comment]
Sorry for responding so late - just tried both the WritingTool_2024-11-17-snapshot.oxt, as well as the newest WritingTool_2024-11-19-snapshot.oxt, but I am also seeing no difference. The issue is still happening for me. 

Still using Aurora, the edited version of Fedora Kionite, still using the Flatpak version, and still on Wayland.

**Comment by securegh** [comment]
> @BobSoft3000 @securegh @YouCanTouCan Please let me know if the problem is resolved.

Yes, it works, I downloaded and tested the `WritingTool_2024-11-20-snapshot.oxt`, thank you for the quick fix, I just had to install java-21-openjdk.x86_64 `sudo dnf install java-21-openjdk.x86_64`
So now, I have access to the settings menu and submenu tools, there might be some needed improvement in for better readability in dark mode. 
![Screenshot from 2024-11-21 02-19-07](https://github.com/user-attachments/assets/5504beb2-6e53-461e-a58c-8b24f271ee0c)

I haven't had the chance to test out the AI features, I will give feedback on that soon. 

Kind regards!

**Comment by YouCanTouCan** [comment]
Still no luck on my end. I'm on WritingTool_2024-11-20-snapshot.oxt, and installed java-21-openjdk.x86_64 through `rpm-ostree install java-21-openjdk.x86_64`, but after a reboot there is still no difference for me, unfortunately. Could this have to do with how the layering of rpm-ostree works?

**Comment by FredKruse** [comment]
After you have installed the new JRE/JDK, you have to select the correct JRE/JDK in LibreOffice under Tools/Options/LibreOffice/Advanced.
Have you done that?

**Comment by securegh** [comment]
> Still no luck on my end. I'm on WritingTool_2024-11-20-snapshot.oxt, and installed java-21-openjdk.x86_64 through `rpm-ostree install java-21-openjdk.x86_64`, but after a reboot there is still no difference for me, unfortunately. Could this have to do with how the layering of rpm-ostree works?
@YouCanTouCan 
The flatpak version of LibreOffice sandbox all of it required dependencies, it won't interact with the OS version of the java OpenJDK. 
You may want to ask on LibreOffice forum how to set up the flatpak version with java-21-openjdk.x86_64
https://ask.libreoffice.org/

**Comment by securegh** [comment]
@YouCanTouCan 

Since you installed java-21-openjdk.x86_64 with rpm-ostree, can you try to manually select the location of the java open jdk installed on your OS.
![image](https://github.com/user-attachments/assets/e1fb521b-6ded-4eaa-a52c-82cbce923fd6)


On my OS, its : `/usr/lib/jvm/java-21-openjdk-21.0.5.0.11-1.fc40.x86_64/`
You may have to give the LibreOffice flatpak access to that location using flatseal.

**Comment by YouCanTouCan** [comment]
Thank you everyone for your help.

I added the location of the java open jdk manually as described, after giving permission through flatseal and finding the location with the help of [this old thread ](https://askubuntu.com/questions/1234328/flatpak-android-studio-cannot-access-usr-lib-jvm-even-after-permission-was-give). However... after applying it and restarting LibreOffice, I still unfortunately see no difference. Clicking the buttons still does nothing.

I am still using WritingTool_2024-11-20-snapshot.oxt

![image](https://github.com/user-attachments/assets/30186a34-6e2f-47f5-a135-815a6a6727d9)

**Comment by YouCanTouCan** [comment]
After a day of use, I ran into the issue that using java-21-openjdk-21.0.5.0.11-1.fc40.x86_64 would unfortunately cause the buttons and windows added by the Zotero addon to also no longer work - in addition to the WritingTool buttons not working. So, I swapped back to the default JRE in the flatpak. 

(side tangent in case someone in the future has this issue: the default JRE no longer showed up in my installed list, but when I deactivated the "use a java runtime environment", restarted my computer, and reopened Libreoffice it prompted me to turn the setting back on. When I did, the flatpak JRE showed up in the list again, and I could swap to it.)

While java-21-openjdk-21.0.5.0.11-1.fc40.x86_64 didn't solve this issue for me anyway, even if it did, I would hesitate to use this as a solution. Installing packages through rpm-ostree already isn't ideal to do on Kionite, and while WritingTool is mostly usable without the buttons working (the biggest loss is the settings menu), Zotero is rendered completely unusable without the buttons. I just wanted to say this in case there was another possible solution that could be pursued without swapping the JRE. If not, then we can continue pursuing this line of fixing it, and I may need to work out the Zotero problem separately. Thank you once again to everyone in this thread for working through this curious problem.

**Comment by FredKruse** [comment]
I am not familiar with Flatpack, but I understand, that it is running in a sandbox. Isn't it possible to submit a support request (open an issue)  to get a full JRE (not only jre-headless) integrated? That would certainly make it easier to install not only WritingTool, but also other LO-extensions that use GUIs.

**Comment by FredKruse** [comment]
I think the problem is solved, so I'm closing the issue. If you still have problems, please open a new issue for the current version.
